### PR TITLE
perl-xml-parser 2.46 (new formula) intltool revision bump

### PR DESCRIPTION
**Pull request by chenrui333**
revision bump to fix couple of linux issues

```
  checking for intltool >= 0.35.0... 0.51.0 found
  checking for intltool-update... /home/linuxbrew/.linuxbrew/opt/intltool/bin/intltool-update
  checking for intltool-merge... /home/linuxbrew/.linuxbrew/opt/intltool/bin/intltool-merge
  checking for intltool-extract... /home/linuxbrew/.linuxbrew/opt/intltool/bin/intltool-extract
  checking for xgettext... /home/linuxbrew/.linuxbrew/opt/gettext/bin/xgettext
  checking for msgmerge... /home/linuxbrew/.linuxbrew/opt/gettext/bin/msgmerge
  checking for msgfmt... /home/linuxbrew/.linuxbrew/opt/gettext/bin/msgfmt
  checking for gmsgfmt... /home/linuxbrew/.linuxbrew/opt/gettext/bin/msgfmt
  checking for perl... /home/linuxbrew/.linuxbrew/opt/perl/bin/perl
  checking for perl >= 5.8.1... 5.38.0
  checking for XML::Parser... configure: error: XML::Parser perl module is required for intltool
```

also update `XML::Parser` to `2.46`

---

relates to:
- https://github.com/Homebrew/homebrew-core/pull/150456
- https://github.com/Homebrew/homebrew-core/pull/149830
- https://github.com/Homebrew/homebrew-core/pull/150244

### CODE REVIEW

**Comment by iMichka** [comment]
Looks good to me in principle, but we have around 15 occurrences of vendored XML::Parser in other formulae. I think we did not use a formula, because in general perl dependencies are vendored. Do you plan to use a formula for the others too?

**Comment by chenrui333** [comment]
> Looks good to me in principle, but we have around 15 occurrences of vendored XML::Parser in other formulae. I think we did not use a formula, because in general perl dependencies are vendored. Do you plan to use a formula for the others too?

We can do a followup, could use the formula on case basis.

File: Formula/p/perl-xml-parser.rb [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  sha256 "d331332491c51cccfb4cb94ffc44f9cd73378e618498d4a37df9e043661c515d"
+  license any_of: ["Artistic-1.0-Perl", "GPL-1.0-or-later"]
+  head "https://github.com/toddr/XML-Parser.git", branch: "master"
+
+  uses_from_macos "perl"
+
+  def install
+    ENV.prepend_create_path "PERL5LIB", libexec/"lib/perl5"
+    system "perl", "Makefile.PL", "INSTALL_BASE=#{libexec}"
+    system "make", "PERL5LIB=#{ENV["PERL5LIB"]}"
```
Comment (Bo98) [inline Formula/p/perl-xml-parser.rb:14]: Could we perhaps install directly into `prefix` rather than `libexec` so we don't need `PERL5LIB` everywhere?

Comment (Bo98) [reply Formula/p/perl-xml-parser.rb:14]: Hmm seems like our Perl might not read from `HOMEBREW_PREFIX/lib/perl5`. What we possibly might want to do is define `vendorlib` to that, and potentially move Perl's own core modules to be in libexec.

Comment (chenrui333) [reply Formula/p/perl-xml-parser.rb:14]: @Bo98 maybe let's get this PR merged and followup with the file location stuff? :)

File: Formula/i/intltool.rb [hunk 2]
```diff
@@ -4,7 +4,7 @@ class Intltool < Formula
   url "https://launchpad.net/intltool/trunk/0.51.0/+download/intltool-0.51.0.tar.gz"
   sha256 "67c74d94196b153b774ab9f89b2fa6c6ba79352407037c8c14d5aeb334e959cd"
   license "GPL-2.0-or-later"
-  revision 1
+  revision 2
```
Comment (MikeMcQuaid) [inline Formula/i/intltool.rb:7]: Does this need `revision i.e. do things break unless it is?

File: Formula/p/perl-xml-parser.rb (hunk 1 above)
Comment (MikeMcQuaid) [reply Formula/p/perl-xml-parser.rb:14]: I agree about this being able to wait until a a future PR.

**Review by MikeMcQuaid — APPROVED** [review]
✅ from me but would like @Bo98 to also 👍🏻 before merge, thanks!

File: Formula/i/intltool.rb (hunk 2 above)
Comment (chenrui333) [reply Formula/i/intltool.rb:7]: yeah, it is already broken for linux build

Comment (MikeMcQuaid) [reply Formula/i/intltool.rb:7]: Already broken i.e. the Linux build is broken before this or without it? Ideally we'd only `revision` if every platform (including users building from source) have to reinstall for a critical change.

File: Formula/p/perl-xml-parser.rb (hunk 1 above)
Comment (Bo98) [reply Formula/p/perl-xml-parser.rb:14]: Sure, though you will have to go round and update every dependent to change `Formula["intltool"].opt_libexec` to `Formula["perl-xml-parser"].opt_libexec`.

**Review by SMillerDev — APPROVED** [review]

**Comment by chenrui333** [comment]
merging.

**Comment by chenrui333** [comment]
gonna rerun the related PRs.

File: Formula/i/intltool.rb (hunk 2 above)
Comment (chenrui333) [reply Formula/i/intltool.rb:7]: it is broken after the perl upgrade.

**Pull request merged**